src/main/java/List.java

```diff
@@ -0,0 +1,315 @@
+import exception.EmptyDateFieldException;
+import exception.EmptyDescriptionException;
+import task.*;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class List {
+    public static final String INVALID_EVENT_INPUT_MESSAGE = "event <event name> /from <start date/time> /end <end date/time>";
+    public static final String INVALID_DEADLINE_INPUT_MESSAGE = "deadline <deadline name> /by <deadline>";
+    public static final String INVALID_TODO_INPUT_MESSAGE = "todo <task name>";
+    public static final String DEADLINE_BY_KEYWORD = "/by";
+    public static final String EVENT_FROM_KEYWORD = "/from";
+    public static final String EVENT_TO_KEYWORD = "/to";
+    private static final String INVALID_MARK_MESSAGE = "mark <task index>";
+    private static final String INVALID_UNMARK_MESSAGE = "unmark <task index>";
+    private static final String INVALID_DELETE_MESSAGE = "delete <task index>";
+
+    private int numItems;
+    private Task[] itemList = new Task[0];
+    ArrayList<Task> itemArrayList = new ArrayList<>(Arrays.asList(itemList));
+
+
+
+    public List() {
+        this.numItems = 0;
+    }
+
+    public int getNumItems() {
+        return itemArrayList.size();
+    }
+
+    public void addItem(String line) {
+        if (isValidEvent(line)) {
+            addEvent(line);
+        } else if (isValidDeadline(line)) {
+            addDeadline(line);
+        } else if (isTodo(line)) {
+            addTodo(line);
+        } else {
+            printInvalidTaskMessage();
+        }
+    }
+
+    private boolean isValidEvent(String line) {
+        return line.startsWith("event") &&
+                line.contains(EVENT_FROM_KEYWORD) &&
+                line.contains(EVENT_TO_KEYWORD);
+    }
+
+    private boolean isValidDeadline(String line) {
+        return line.startsWith("deadline") && line.contains(DEADLINE_BY_KEYWORD);
+    }
+
+    private boolean isTodo(String line) {
+        return line.startsWith("todo");
+    }
+
+    private static void printInvalidTaskMessage() {
+        System.out.println("\tInvalid command format:" + System.lineSeparator() + "\t\t" + INVALID_TODO_INPUT_MESSAGE
+                + System.lineSeparator() + "\t\t" + INVALID_DEADLINE_INPUT_MESSAGE + System.lineSeparator() + "\t\t"
+                + INVALID_EVENT_INPUT_MESSAGE + System.lineSeparator() + "\t\t"
+                + INVALID_MARK_MESSAGE + System.lineSeparator() + "\t\t"
+                + INVALID_UNMARK_MESSAGE + System.lineSeparator() + "\t\t"
+                + INVALID_DELETE_MESSAGE);
+    }
```

Comment:
Consider using switch case, as the different values that you are expecting are predetermined